namespaces/resource-namespace.md

```diff
@@ -0,0 +1,328 @@
+
+## Content
+
+<!-- TOC depth:3 withLinks:1 updateOnSave:0 -->
+- [Resource Namespace](#resource-namespace)
+    - [Content](#content)
+    - [About this Document](#about-this-document)
+    - [Supporting Data Types](#supporting-data-types)
+        - [Href (string)](#href-string)
+        - [Templated Href (string)](#templated-href-string)
+        - [Href Variables (Object Type)](#href-variables-object-type)
+        - [Data Structure (MSON Element)](#data-structure-mson-element)
+        - [Asset (Element)](#asset-element)
+    - [Resource Elements](#resource-elements)
+        - [Resource (Element)](#resource-element)
+        - [Transition (Element)](#transition-element)
+    - [Protocol-specific Elements](#protocol-specific-elements)
+        - [HTTP Transaction (Element)](#http-transaction-element)
+        - [HTTP Headers (Array Type)](#http-headers-array-type)
+        - [HTTP Message Payload (Element)](#http-message-payload-element)
+        - [HTTP Request Message (HTTP Message Payload)](#http-request-message-http-message-payload)
+        - [HTTP Response Message (HTTP Message Payload)](#http-response-message-http-message-payload)
+
+<!-- /TOC -->
+
+## About this Document
+
+This document conforms to [RFC 2119][], which says:
+
+> The key words “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+[MSON][] is used throughout this document.
+
+## Supporting Data Types
+
+### Href (string)
+
+The value of the `Href` type  SHOULD be resolved as a URI-Reference per [RFC 3986][] and MAY be a relative reference to a URI.
+The value of the `Href` type MUST NOT be a URI Template.
+
+### Templated Href (string)
+
+The value of `Templated Href` type is to be used as a URI Template, as defined in [RFC 6570][].
+The value of the `Templated Href` type is a template used to determine the target URI of the related resource or transition.
+The value of the `Templated Href` type SHOULD be resolved as a URI-Reference per [RFC 3986][] and MAY be a relative reference to a URI.
+
+### Href Variables (Object Type)
+
+The definition is a MSON namespace `Object Type` where keys are respective URI Template variables.
+
+#### Properties
+
+- `element`: hrefVariables (string, fixed)
+
+### Data Structure (MSON Element)
+
+Data structure definition using MSON namespace elements.
+
+#### Properties
+
+- `element`: dataStructure (string, fixed)
+
+### Asset (Element)
+
+Arbitrary data asset.
+
+#### Properties
+
+- `element`: asset (string, fixed)
+- `meta`
+    - `class` (array, fixed)
+        - (enum[string])
+            - messageBody - Asset is an example of message-body
+            - messageBodySchema - Asset is an schema for message-body
+- `attributes` (object)
+    - `contentType` (string) - Optional media type of the asset
+    - `href` (Href) - Link to the asset
+- `content` (string) - A textual representation of the asset
```

Comment:
Missing trailing full stop on a few of the properties such as the above. Only pointing out because some have a full-stop and it's inconsistent. But not really a problem.

> - `statusCode` (number) - HTTP response status code.